kuka_sunrise_fri_driver/include/kuka_sunrise_fri_driver/robot_manager_node.hpp

```diff
@@ -51,28 +56,44 @@ class RobotManagerNode : public kuka_drivers_core::ROS2BaseLCNode
   virtual rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn on_deactivate(
     const rclcpp_lifecycle::State &);
 
-  bool activateControl();
-  bool deactivateControl();
-
 private:
   std::shared_ptr<FRIConnection> fri_connection_;
-  std::unique_ptr<ConfigurationManager> configuration_manager_;
-  rclcpp::Client<std_srvs::srv::Trigger>::SharedPtr set_parameter_client_;
   rclcpp::Client<controller_manager_msgs::srv::SetHardwareComponentState>::SharedPtr
     change_hardware_state_client_;
   rclcpp::Client<controller_manager_msgs::srv::SwitchController>::SharedPtr
     change_controller_state_client_;
   rclcpp::CallbackGroup::SharedPtr cbg_;
-  rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Bool>::SharedPtr
-    command_state_changed_publisher_;
+  rclcpp::CallbackGroup::SharedPtr event_cbg_;
   std::shared_ptr<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Bool>> is_configured_pub_;
   std_msgs::msg::Bool is_configured_msg_;
-  std::string controller_name_;
+  rclcpp::Publisher<kuka_driver_interfaces::msg::FriConfiguration>::SharedPtr fri_config_pub_;
+  rclcpp::Publisher<std_msgs::msg::UInt32>::SharedPtr control_mode_pub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr joint_imp_pub_;
+  rclcpp::Subscription<std_msgs::msg::UInt8>::SharedPtr event_subscriber_;
+  std_msgs::msg::UInt32 control_mode_msg_;
+
+  int receive_multiplier_ = 0;
+  int send_period_ms_ = 0;
   std::string robot_model_;
+  std::string joint_pos_controller_name_;
+  std::string joint_torque_controller_name_;
+  std::vector<double> joint_stiffness_ = std::vector<double>(7, 100.0);
+  std::vector<double> joint_damping_ = std::vector<double>(7, 0.7);
 
-  void handleControlEndedError();
-  void handleFRIEndedError();
+  std::string GetControllerName();
+  bool onControlModeChangeRequest(int control_mode);
   bool onRobotModelChangeRequest(const std::string & robot_model);
+  bool onSendPeriodChangeRequest(int send_period);
+  bool setReceiveMultiplier(int receive_multiplier);
+  bool onReceiveMultiplierChangeRequest(const int & receive_multiplier);
+  bool ValidateIPAdress(const std::string & controller_ip) const;
```

Comment:
would not that cause problems? as I see most packages use only C++14, and for this 17 is needed
I will try to invvestigate whether there is a requirement about which C++ version should be supported

Reply:
done for `ValidateIPAdress` and `onControllerNameChangeRequest`, should I convert anything else?

Reply:
I think it's fine now. But if you really want to you can replace a few return values too!

Reply:
`GetControllerName` is used in a string vector, so that should be converted back to string
`onRobotModelChangeRequest` uses + for string addition, another conversion would be needed, so I think I will leave those for now